syncano/models/manager.py

```diff
@@ -0,0 +1,251 @@
+        return self.get(*args, **kwargs)
+
+    def get_or_create(self, *args, **kwargs):
+        defaults = deepcopy(kwargs.pop('defaults', {}))
+        try:
+            instance = self.get(*args, **kwargs)
+        except self.model.DoesNotExist:
+            defaults.update(self.properties)
+            defaults.update(kwargs)
+            instance = self.create(**defaults)
+        return instance
+
+    def delete(self, *args, **kwargs):
+        self.method = 'DELETE'
+        self.endpoint = 'detail'
+        self._filter(*args, **kwargs)
+        return self.request()
+
+    def update(self, *args, **kwargs):
+        self.method = 'POST'
+        self.endpoint = 'detail'
+        self.data = kwargs.pop('data')
+        self._filter(*args, **kwargs)
+        return self.request()
+
+    def update_or_create(self, *args, **kwargs):
+        data = deepcopy(kwargs.get('data', {}))
+        try:
+            instance = self.update(*args, **kwargs)
+        except self.model.DoesNotExist:
+            data.update(self.properties)
+            data.update(kwargs)
+            instance = self.create(**data)
+        return instance
+
+    # List actions
+
+    def all(self, *args, **kwargs):
+        self._limit = None
+        return self.list(*args, **kwargs)
+
+    def list(self, *args, **kwargs):
+        self.method = 'GET'
+        self.endpoint = 'list'
+        self._filter(*args, **kwargs)
+        return self._clone()
+
+    def page_size(self, value):
+        if not value or not isinstance(value, six.integer_types):
+            raise SyncanoValueError('page_size value needs to be an int.')
+
+        self.query['page_size'] = value
+        return self._clone()
+
+    def limit(self, value):
+        if not value or not isinstance(value, six.integer_types):
+            raise SyncanoValueError('Limit value needs to be an int.')
+
+        self._limit = value
+        return self._clone()
+
+    def order_by(self, field):
+        if not field or not isinstance(field, six.string_types):
+            raise SyncanoValueError('Order by field needs to be a string.')
+
+        self.query['order_by'] = field
+        return self._clone()
+
+    def raw(self):
+        self._serialize = False
+        return self._clone()
+
+    # Other stuff
+
+    def contribute_to_class(self, model, name):
+        setattr(model, name, ManagerDescriptor(self))
+
+        self.model = model
+        if hasattr(model._meta, 'connection') and model._meta.connection:
+            self.connection = model._meta.connection
+
+        if not self.name:
+            self.name = name
+
+    def _filter(self, *args, **kwargs):
+        if args and self.endpoint:
+            properties = self.model._meta.get_endpoint_properties(self.endpoint)
+            mapped_args = {k: v for k, v in zip(properties, args)}
+            self.properties.update(mapped_args)
+        self.properties.update(kwargs)
+
+    def _clone(self):
+        # Maybe deepcopy ?
```

Comment:
Maybe it's maybelline? ;) Deepcopy would definitely make code shorter, but what about memory footprint? We don't need deep copies of everything.